cfme/tests/middleware/test_middleware_deployment.py

```diff
@@ -1,33 +1,48 @@
-from random import sample
-
 import pytest
+from cfme.middleware import get_random_list
 from cfme.middleware.deployment import MiddlewareDeployment
+from cfme.middleware.server import MiddlewareServer
 from utils import testgen
 from utils.version import current_version
 
 pytestmark = [
-    pytest.mark.usefixtures('single_middleware_provider'),
+    pytest.mark.usefixtures('setup_provider'),
```

Comment:
OK then I misunderstood what @psav told me during our previous sessions. I thought that cfme_tests allow to have only single middleware provider. That is why I added fixture to clean all providers and leave only one.
This changes a lot.
We can have several providers in yaml, create them in fixtures, and "Provider" objects have in tests, not taken from UI via "get_crud(get_provider_key(provider_name))" but have them provided directly from config files.